### ISSUE TITLE
intersect('^5.0.0', '^4.0.1') wrongfully claims they intersect at 5.0.0

### ISSUE THREAD

**Issue by heikomat** [snyamathi/semver-intersect#7]
comparing `^4.0.1`  and `^5.0.0` works as expected
```
> intersect('^4.0.1', '^5.0.0')
Error: Range >=5.0.0 is not compatible with <5.0.0
```

but switching the numbers makes semver-intersect wrongfully claim that they intersect at exactly 5.0.0:
```
> intersect('^5.0.0', '^4.0.1')
'5.0.0'
```

**Comment by snyamathi** [comment]
Thanks @heikomat I've added a test in https://github.com/snyamathi/semver-intersect/compare/err?expand=1 which does fail with the conditions that you've documented here.  I'll look into a fix tomorrow if I can.

Do feel free to provide a PR if you have some bandwidth before then, but I appreciate the report!

**Comment by snyamathi** [comment]
Released in v1.3.1

**Comment by heikomat** [comment]
@snyamathi Thank you again for that fast fix :)
